corehq/ex-submodules/casexml/apps/case/util.py

```diff
@@ -21,6 +21,9 @@
 
 
 def validate_phone_datetime(datetime_string, none_ok=False, form_id=None):
+    if isinstance(datetime_string, datetime.datetime):
```

Comment:
got it. so just to make sure I understand, the main issue is that when the form is being parsed for the first time during submission the datetimes are not converted from strings when they enter `_extract_case_blocks`, but once they are coming out of the database they are?

If that's right this approach sounds good to me. Might be worth a quick comment explaining it above, but otherwise this seems good to merge. Just holding off to make sure I've understood it.